pkg/util/helper/taint_test.go

```diff
@@ -357,3 +359,209 @@ func TestGetNoExecuteTaints(t *testing.T) {
 		})
 	}
 }
+
+func TestGetMinTolerationTime(t *testing.T) {
+	now := metav1.Now()
+	later := metav1.NewTime(time.Now().Add(5000))
+	type args struct {
+		noExecuteTaints []corev1.Taint
+		usedTolerations []corev1.Toleration
+	}
+	tests := []struct {
+		name string
+		args args
+		want time.Duration
+	}{
+		{
+			name: "test no noExecute taints",
+			args: args{
+				noExecuteTaints: nil,
+				usedTolerations: []corev1.Toleration{
+					{
+						Key:               "key",
+						Operator:          "Equal",
+						Value:             "value",
+						Effect:            "NoExecute",
+						TolerationSeconds: nil,
+					},
+				},
+			},
+			want: -1,
+		},
+		{
+			name: "test no tolerations",
+			args: args{
+				noExecuteTaints: []corev1.Taint{
+					{
+						Key:       "key",
+						Value:     "value",
+						Effect:    "NoExecute",
+						TimeAdded: nil,
+					},
+				},
+				usedTolerations: nil,
+			},
+			want: 0,
+		},
+		{
+			name: "test get minimal toleration time",
+			args: args{
+				noExecuteTaints: []corev1.Taint{
+					{
+						Key:       "key",
+						Value:     "value",
+						Effect:    "NoExecute",
+						TimeAdded: nil,
+					},
+					{
+						Key:       "key",
+						Value:     "value",
+						Effect:    "NoExecute",
+						TimeAdded: &now,
+					},
+					{
+						Key:       "key",
+						Value:     "value",
+						Effect:    "NoExecute",
+						TimeAdded: &later,
+					},
+				},
+				usedTolerations: []corev1.Toleration{
+					{
+						Key:               "key",
+						Operator:          "Equal",
+						Value:             "value",
+						Effect:            "NoExecute",
+						TolerationSeconds: nil,
+					},
+					{
+						Key:               "key",
+						Operator:          "Equal",
+						Value:             "value",
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(10000),
+					},
+					{
+						Key:               "key",
+						Operator:          "Equal",
+						Value:             "value",
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(20000),
+					},
+				},
+			},
+			want: 9999999999999,
```

Comment:
I test the case failed on my local side:
```
taint_test.go:459: GetMinTolerationTime() = 10000000, want 9999999
```

I think it's best that we don't use the current time when we use the time: 

```go
# line 364
now := metav1.Now()
```

Reply:
ok